File: 2/solution_2.py
def fibonacci_evens():
    greatest = 89
    second = 55
    sum = 44
    while greatest <= 4000000:
        new_greatest = greatest + second
        if new_greatest % 2 == 0:
            sum += new_greatest
        second = greatest
        greatest = new_greatest
    return sum 

print(fibonacci_evens())

File: 2/solution_2.cpp
#include <iostream>
using namespace std;

int fibonacciEvenSum(){
    unsigned int greatest = 89;
    unsigned int second = 55;
    unsigned int sum = 44;

    while(greatest <= 4000000){
        unsigned int newGreatest = greatest + sum;
        if(newGreatest % 2 == 0){
            sum += newGreatest;
        }
        second = greatest;
        greatest = newGreatest;
    }
    return sum;
};

int main(){
    cout << fibonacciEvenSum();
};

